avocado/utils/cpu.py

```diff
@@ -374,3 +376,49 @@ def get_pid_cpus(pid):
         except IOError:
             continue
     return list(cpus)
+
+
+def _is_smt():
+
+    """
+    Return True/False Based on if system has hyperthreading enabled
+    :return: True/False
+    :rtype: boolean
+    """
+
+    total_logical_cpus = 0
+    total_physical_cpus = 0
+    total_cores = 0
+    logical_cpus = {}
+    physical_cpus = {}
+    cores = {}
+    if get_cpu_vendor_name().startswith('power'):
+        try:
+            if b'is not SMT capable' in process.system_output("ppc64_cpu --smt"):
+                return False
+            return True
+        except process.CmdError as err:
+            logging.error("Unable to get hyperthreading info  %s", err)
+    else:
+        for line in open('/proc/cpuinfo').readlines():
+            if re.match('processor', line):
+                cpu = int(line.split()[2])
+            if cpu not in logical_cpus:
```

Comment:
why do we need `logical_cpus`, `physical_cpus` and `cores` dict, it not getting used anywhere else, apart from assigning to None..?

I assume you want to count once per `physical id	: 0`, if then this would do

```
$cat /proc/cpuinfo |grep "core id"|sort -u|wc -l
2
$ cat /proc/cpuinfo |grep "physical id"|sort -u|wc -l
1
$ cat /proc/cpuinfo |grep "processor"|sort -u|wc -l
4
```

Reply:
i thought that as well , but i don't want to use 'grep' inside python , let me see if i can get that using `re` module